pkgs/development/python-modules/fmpy/default.nix

```diff
@@ -0,0 +1,213 @@
+  };
+
+  nativeBuildInputs = [
+    cmake
+    hatchling
+  ];
+
+  propagatedBuildInputs = [
+    pyqtgraph
+    pyside6
+    attrs
+    lark
+    lxml
+    msgpack
+    numpy
+    scipy
+    pytz
+    dask
+    requests
+    matplotlib
+    pyqtgraph
+    notebook
+    plotly
+    rpclib
+    cvode
+    pyside6
+    jinja2
+    flask
+    dash
+    dash-bootstrap-components
+  ];
+
+  dontUseCmakeConfigure = true;
+  dontUseCmakeBuildDir = true;
+
+  patches = [
+    ./0001-gui-override-Qt6-libexec-path.patch
+  ];
+
+  postPatch =
+    ''
+      substitute ${./libraries.py} ./src/fmpy/sundials/libraries.py \
+          --subst-var-by cvode ${cvode}
+      substituteInPlace src/fmpy/gui/__init__.py \
+          --replace-fail "@QT6_LIBEXEC_DIR@" "${qt6.qtbase}/libexec/"
+
+      # Make forced includes of other systems' artifacts (otherwise vendored upstream from CI) optional instead.
+      sed --in-place 's/force-include/sourced/g' pyproject.toml
+    '';
+
+  # Don't run upstream build scripts as they are too specialized.
+  # cvode is already built, so we only need to build native binaries.
+  preBuild =
+    ''
+      cmakeFlags="-S native/src -B native/src/build -D CVODE_INSTALL_DIR=${cvode}"
+      cmakeConfigurePhase
+      cmake --build native/src/build --config Release
+    ''
+    + lib.optionalString (enableRemoting && stdenv.isLinux) ''
+      # reimplementation of native/build_remoting.py
+      cmakeFlags="-S native/remoting -B remoting/linux64 -D RPCLIB=${rpclib}"
+      cmakeConfigurePhase
+      cmake --build remoting/linux64 --config Release
+    ''
+    # C.f. upstream build-wheel CI job
+    + ''
+      python native/copy_sources.py
+
+      # reimplementation of native/compile_resources.py
+      pushd src/
+      python -c "from fmpy.gui import compile_resources; compile_resources()"
+      popd
+    '';
+
+  pythonImportsCheck = [
+    "fmpy"
+    "fmpy.cross_check"
+    "fmpy.cswrapper"
+    "fmpy.examples"
+    "fmpy.fmucontainer"
+    "fmpy.logging"
+    "fmpy.gui"
+    "fmpy.gui.generated"
+    "fmpy.ssp"
+    "fmpy.sundials"
+  ];
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+
+  # Simulate reference FMUs from
+  # <https://github.com/modelica/Reference-FMUs>.
+  passthru.tests.ref-fmus-simulate = runCommand "${pname}-simulate-test" { } ''
+    mkdir $out
+
+    # Just check that the execution passes; don't verify any numerical
+    # results, but save them in case of future or manual check use.
+    #
+    # NB(find ! -name): Clocks.fmu is marked TODO upstream and is of a
+    # FMI type that FMPy doesn't support currently (ModelExchange)
+    for fmu in $(find ${fmi-reference-fmus}/*.fmu ! -name "Clocks.fmu"); do
+      name=$(basename $fmu)
+      echo "--- START $name ---"
+      ${fmpy}/bin/fmpy simulate $fmu \
+        --fmi-logging \
+        --output-file $out/$name.csv \
+        | tee $out/$name.out
+    done
+  '';
+
+  passthru = {
```

Comment:
You'd probably laugh, but this statement actually deletes the `passthru.tests`! Please define the tests here below.

Reply:
Good catch!